android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java

```diff
@@ -218,10 +224,54 @@ public View onCreateView(LayoutInflater layoutInflater, ViewGroup container,
   @Override
   public void onResume() {
     super.onResume();
-    getActivity().getActionBar().setTitle(R.string.title_nearby_beacons);
-    getActivity().getActionBar().setDisplayHomeAsUpEnabled(false);
-    getListView().setVisibility(View.INVISIBLE);
-    mDiscoveryServiceConnection.connect(true);
+    ensureLocationPermissionIsEnabled();
+  }
+
+  @Override
+  public void onRequestPermissionsResult(int requestCode,
```

Comment:
Should this just be part of the main activity rather than the fragment? (this way we wouldn't have to do another check in the config fragment)